MediaBrowser.MediaEncoding/Encoder/MediaEncoder.cs

```diff
@@ -849,14 +849,89 @@ protected virtual void Dispose(bool dispose)
             }
         }
 
+        /// <inheritdoc />
         public Task ConvertImage(string inputPath, string outputPath)
         {
             throw new NotImplementedException();
         }
 
+        /// <inheritdoc />
         public IEnumerable<string> GetPrimaryPlaylistVobFiles(string path, IIsoMount isoMount, uint? titleNumber)
         {
-            throw new NotImplementedException();
+            // min size 300 mb
+            const long MinPlayableSize = 314572800;
+
+            var root = isoMount != null ? isoMount.MountedPath : path;
+
+            // Try to eliminate menus and intros by skipping all files at the front of the list that are less than the minimum size
+            // Once we reach a file that is at least the minimum, return all subsequent ones
+            var allVobs = _fileSystem.GetFiles(root, true)
+                .Where(file => string.Equals(file.Extension, ".vob", StringComparison.OrdinalIgnoreCase))
+                .OrderBy(i => i.FullName)
+                .ToList();
+
+            // If we didn't find any satisfying the min length, just take them all
+            if (allVobs.Count == 0)
+            {
+                _logger.LogWarning("No vobs found in dvd structure.");
+                return Enumerable.Empty<string>();
+            }
+
+            if (titleNumber.HasValue)
+            {
+                var prefix = string.Format("VTS_0{0}_", titleNumber.Value.ToString(CultureInfo.InvariantCulture));
```

Comment:
hmm, shouldn't it be done like (from memory, untested):
```cs
var prefix = string.Format(CultureInfo.InvariantCulture, "VTS_0{0}_", titleNumber);
```
?
Also it would probably break for non-single-digit titles...